Src/GoogleApis.Auth.DotNet4.Tests/OAuth2/DefaultCredentialProvider/DefaultCredentialProviderTests.cs

```diff
@@ -0,0 +1,270 @@
+        }
+
+        public void SetFileContents(string filePath, string contents)
+        {
+            fileContents[filePath] = contents;
+        }
+    }
+
+    /// <summary>Tests for <see cref="Google.Apis.Auth.OAuth2.DefaultCredentialProvider"/>.</summary>
+    [TestFixture]
+    public class DefaultCredentialProviderTests
+    {
+        private static readonly string WellKnownCredentialFilePath = Path.Combine(AppDataValue, "gcloud", "application_default_credentials.json");
+        private static readonly string WellKnownCredentialFilePathUnix = Path.Combine(HomeValue, ".config", "gcloud", "application_default_credentials.json");
+
+        private const string AppDataValue = "AppDataEnvVarValue";
+        private const string HomeValue = "HomeEnvVarValue";
+        private const string CredentialEnvironmentVariable = "GOOGLE_APPLICATION_CREDENTIALS";
+        private const string DummyUserCredentialFileContents = @"{
+""client_id"": ""CLIENT_ID"",
+""client_secret"": ""CLIENT_SECRET"",
+""refresh_token"": ""REFRESH_TOKEN"",
+""type"": ""authorized_user""}";
+        private const string DummyServiceAccountCredentialFileContents = @"{
+""private_key_id"": ""PRIVATE_KEY_ID"",
+""private_key"": ""-----BEGIN PRIVATE KEY-----
+MIICdgIBADANBgkqhkiG9w0BAQEFAASCAmAwggJcAgEAAoGBAJJM6HT4s6btOsfe
+2x4zrzrwSUtmtR37XTTi0sPARTDF8uzmXy8UnE5RcVJzEH5T2Ssz/ylX4Sl/CI4L
+no1l8j9GiHJb49LSRjWe4Yx936q0Xj9H0R1HTxvjUPqwAsTwy2fKBTog+q1frqc9
+o8s2r6LYivUGDVbhuUzCaMJsf+x3AgMBAAECgYEAi0FTXsu/zRswAUGaViQiHjrL
+uU65BSHXNVjV/2fLNEKnGWGqpli68z1IXY+S2nwbUak7rnGsq9/0F6jtsW+hZbLk
+KXUOuuExpeC5Kd6ngWX/f2jqmhlUabiQijU9cVk7pMq8EHkRtvlosnMTUAEzempu
+QUPwn1PZHhmJkBvZ4lECQQDCErrxl+e3BwUDcS0yVEEmCNSG6xdXs2878b8rzbe7
+3Mmi6SuuOLi3PU92J+j+f/MOdtYrk13mEDdYmd5dhrt5AkEAwPvDEsDT/W4y4h5n
+gv1awGBA5aLFE1JNWM/Gwn4D1cGpEDHKFREaBtxMDCASpHJuw8r7zUywpKhmBZcf
+GS37bwJANdSAKfbafLfjuhqwUJ9yGpykZm/a36aTmerp/bpn1iHdg+RtCzwMcDb/
+TWSwibbvsflgWmHbz657y4WSWhq+8QJAWrpCNN/ZCk2zuGDo80lfUBAwkoVat8G6
+wWU1oZyS+vzIGef+hLb8kHsjeZPej9eIwZ39kcBbT54oELrCkRjwGwJAQ8V2A7lT
+ZUp8AsbVqF6rbLiiUfJMo2btGclQu4DEVyS+ymFA65tXDLUuR9EDqJYdqHNZJ5B8
+4Z5p2prkjWTLcA\u003d\u003d
+-----END PRIVATE KEY-----"",
+""client_email"": ""CLIENT_EMAIL"",
+""client_id"": ""CLIENT_ID"",
+""type"": ""service_account""}";
+        private const string BrokenPkcs8KeyServiceAccountCredentialFileContents = @"{
+""private_key_id"": ""PRIVATE_KEY_ID"",
+""private_key"": ""-----BEGIN PRIVATE KEY-----
+MIICdgIBADANBgkqhkiG9w0BAQEFAASCAmAwggJcAgEAAoGBAJJM6HT4s6btOsfe
+-----END PRIVATE KEY-----"",
+""client_email"": ""CLIENT_EMAIL"",
+""client_id"": ""CLIENT_ID"",
+""type"": ""service_account""}";
+
+        private MockDefaultCredentialProvider credentialProvider;
+
+        [SetUp]
+        public void SetUp()
+        {
+            credentialProvider = new MockDefaultCredentialProvider();
+        }
+
+        #region UserCredential
+
+        [Test]
+        public async Task GetDefaultCredential_UserCredential_FromEnvironmentVariable()
```

Comment:
I am leaning against it. Usually when you add comments to your test, you are just duplicating the information that is available in the test name already and that later becomes pain for maintenance (and you often end up with mismatch in test name and its comment after some period of time).

This test is called GetDefaultCredential_UserCredential_FromEnvironmentVariable,
so it tests the GetDefaultCredential method in the case where it should return UserCredential that is located based on an environment variable. I can still add a comment but I don't feel it would actually help the readability in the long run.